File: NXNavigationExtensionDemo/Feature/Basic/Controllers/ViewController09_FullScreen.m
//
//  ViewController09_FullScreenViewController09_FullScreen.m
//  NXNavigationExtensionDemo
//
//  Created by Leo Lee on 2020/10/26.
//

#import <NXNavigationExtension/NXNavigationExtension.h>

#import "ViewController09_FullScreen.h"

@interface ViewController09_FullScreen ()

@end

@implementation ViewController09_FullScreen

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = self.randomColor;
}

- (UIStatusBarStyle)preferredStatusBarStyle {
    return UIStatusBarStyleLightContent;
}

- (UIColor *)nx_barTintColor {
    return [UIColor whiteColor];
}

- (NSDictionary<NSAttributedStringKey,id> *)nx_titleTextAttributes {
    return @{NSForegroundColorAttributeName: [self nx_barTintColor]};
}

- (UIColor *)nx_navigationBarBackgroundColor {
    return self.randomColor;
}

- (UIColor *)nx_shadowImageTintColor {
    return [UIColor clearColor];
}

@end

File: Example/NXNavigationExtensionSwiftDemo/Basic/ViewController09_FullScreen.swift
//
//  ViewController09_FullScreen.swift
//  NXNavigationExtensionSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit
import NXNavigationExtension

class ViewController09_FullScreen: BaseViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        view.backgroundColor = randomColor
    }
    
    override var preferredStatusBarStyle: UIStatusBarStyle {
        return .lightContent
    }
    
    override var nx_barTintColor: UIColor? {
        return .white
    }
    
    override var nx_titleTextAttributes: [NSAttributedString.Key : Any]? {
        return [NSAttributedString.Key.foregroundColor: UIColor.white]
    }
    
    override var nx_navigationBarBackgroundColor: UIColor? {
        return randomColor
    }
    
    override var nx_shadowImageTintColor: UIColor? {
        return .clear
    }

}
